extensions-core/s3-extensions/src/main/java/io/druid/storage/s3/S3TimestampVersionedDataFinder.java

```diff
@@ -65,23 +68,28 @@ public URI getLatestVersion(final URI uri, final @Nullable Pattern pattern)
             final S3Coords coords = new S3Coords(checkURI(uri));
             long mostRecent = Long.MIN_VALUE;
             URI latest = null;
-            S3Object[] objects = s3Client.listObjects(coords.bucket, coords.path, null);
-            if (objects == null) {
-              return null;
-            }
-            for (S3Object storageObject : objects) {
-              storageObject.closeDataInputStream();
-              String keyString = storageObject.getKey().substring(coords.path.length());
+            final Iterator<S3ObjectSummary> objectSummaryIterator = S3Utils.objectSummaryIterator(
+                s3Client,
+                coords.bucket,
+                coords.path,
+                MAX_LISTING_KEYS
+            );
+            while (objectSummaryIterator.hasNext()) {
+              final S3ObjectSummary objectSummary = objectSummaryIterator.next();
+              // TODO: what is going on here?
+              String keyString = objectSummary.getKey().substring(coords.path.length());
```

Comment:
it is taking the path beyond the "search" prefix, and checking the regex to see if it matches the remainder (after the prefix)

Reply:
Thanks.